operator/apis/lifecycle/v1alpha3/keptntaskdefinition_types.go

```diff
@@ -125,3 +132,50 @@ type KeptnTaskDefinitionList struct {
 func init() {
 	SchemeBuilder.Register(&KeptnTaskDefinition{}, &KeptnTaskDefinitionList{})
 }
+
+func (d KeptnTaskDefinition) SpecExists() bool {
+	return d.IsJSSpecDefined() || d.IsContainerSpecDefined()
+}
+
+func (d KeptnTaskDefinition) IsJSSpecDefined() bool {
+	return d.Spec.Function != nil
+}
+
+func (d KeptnTaskDefinition) IsContainerSpecDefined() bool {
+	return d.Spec.Container != nil
+}
+
+func (d KeptnTaskDefinition) IsVolumeMountPresent() bool {
+	return d.IsContainerSpecDefined() && d.Spec.Container.VolumeMounts != nil && len(d.Spec.Container.VolumeMounts) > 0
+}
+
+func (d KeptnTaskDefinition) GetVolumeSource() *v1.EmptyDirVolumeSource {
+	if d.IsContainerSpecDefined() {
+		quantity, ok := d.Spec.Container.Resources.Limits["memory"]
+		if ok {
+			return &v1.EmptyDirVolumeSource{
+				SizeLimit: &quantity,
+			}
+		}
+	}
+
+	return &v1.EmptyDirVolumeSource{
+		// Default 50% of the memory of the node, max 1Gi
+		SizeLimit: resource.NewQuantity(1, resource.Format("Gi")),
+		Medium:    v1.StorageMedium("Memory"),
+	}
+}
+
+func (d KeptnTaskDefinition) GenerateVolumes() []v1.Volume {
+	if !d.IsVolumeMountPresent() {
+		return []v1.Volume{}
+	}
+	return []v1.Volume{
```

Comment:
why do we return an array of a single element?

Reply:
We are implementing the interface, that is common for python and js builders as well. There there is a possibility to return and create multiple volumes, therefore here we need to return an array, although there is always max 1 volume

Reply:
here is already a ticket for refactoring the interface and polishing the code https://github.com/keptn/lifecycle-toolkit/issues/1500

Reply:
Initially, I refactored the interface to return a slice of volumes since we wanted to let users define as many volumes in the container as they like, but there is no need in my opinion now, since both js and container runtime only allow exactly one volume

Reply:
I mean the interface should be refactored to support 2 functions: one for returning the container and the second one for returning volume, should we do it right away now as part of this PR or in a follow-up? If I am going to refactor it to return one volume, I can directly do the full refactoring, WDYT @thisthat @RealAnna ?